plugins/inputs/snmp_trap/snmp_trap.go

```diff
@@ -116,6 +117,9 @@ func (s *SnmpTrap) Start(acc telegraf.Accumulator) error {
 	s.listener = gosnmp.NewTrapListener()
 	s.listener.OnNewTrap = makeTrapHandler(s)
 	s.listener.Params = gosnmp.Default
+	s.udp = &gosnmp.GoSNMP{
```

Comment:
s.udp is not used somewhere?

Reply:
This section is setting GoSNMP options.

Reply:
Are you sure? It seems like its never reverenced anymore in the code..

Reply:
since snmp_trap is an event-based input plugin, start is called by telegraf during initialization and never called again.

Reply:
In telegraf no, but in GoSNMP yes.

Reply:
So did you actually check if the setting was set?

Actually, I don't think it even makes sense with snmp traps, because you are not receiving a packet that came as response to any request..

Reply:
@MyaLongmire ?

Reply:
@MyaLongmire ?

Reply:
The setting is set and since traps use udp i bet it also needs the setting

Reply:
Please enlighten me how it would have any effect in traps?

Reply:
I'm sorry @Hipska, I misinterpreted this [comment](https://github.com/influxdata/telegraf/issues/11278#issuecomment-1178203096). I have since removed the option for traps. Thank you for your continued patience as I learn snmp